plugin/storage/es/spanstore/reader_test.go

```diff
@@ -132,14 +131,33 @@ func withArchiveSpanReader(readAlias bool, fn func(r *spanReaderTest)) {
 var _ spanstore.Reader = &SpanReader{} // check API conformance
 
 func TestNewSpanReader(t *testing.T) {
-	client := &mocks.Client{}
-	reader := NewSpanReader(SpanReaderParams{
-		Client:         client,
-		Logger:         zap.NewNop(),
-		MaxSpanAge:     0,
-		MetricsFactory: metrics.NullFactory,
-		IndexPrefix:    ""})
-	assert.NotNil(t, reader)
+	tests := []struct {
+		name       string
+		params     SpanReaderParams
+		maxSpanAge time.Duration
+	}{
+		{
+			name: "no rollover",
+			params: SpanReaderParams{
+				MaxSpanAge: time.Hour * 72,
+			},
+			maxSpanAge: time.Hour * 72,
+		},
+		{
+			name: "rollover enabled",
+			params: SpanReaderParams{
+				MaxSpanAge:          time.Hour * 72,
+				UseReadWriteAliases: true,
+			},
+			maxSpanAge: time.Hour * 24 * 365 * 100,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			reader := NewSpanReader(test.params)
+			require.NotNil(t, reader)
```

Comment:
i think you're missing the assertion on `test.maxSpanAge` here (I would rename it `test.wantMaxSpanAge` to make it clear it's the expected value)?

Reply:
doh, fixed